### PR TITLE
docs(concepts): shared modules allow multiple versions in a single build

### LINKED ISSUE #3778: Document webpack change: Refactor shared modules to allows multiple version in a single build
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/11043

---

**What kind of change does this PR introduce?**
refactor, bugfixes

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
ModuleFederationPlugin

* relative and absolute request in shared
  * will always be provided, even if not used
  * will resolve relative to `config.context`
  * do not use a `requiredVersion` by default
* module requests in shared
  * are only provided when they are used
  * will match all used equal module requests in your build
  * will provide all matching modules
  * will extract `requiredVersion` from package.json **at this position** in the graph
  * could provide and consume multiple different version when you have nested node_modules
* module requests with tailing `/` in shared
  * will match all module requests with this prefix.
* new `packageName` option to choose package name to look for a `requiredVersion`. For module requests automatically inferred by default.
* `requiredVersion` can now be `false` when automatic infer should be disabled.
* See also the `webpack/examples/module-federation` example for a lot of explanations and examples.

### PR DESCRIPTION
**Pull request by jamesgeorge007**
fixes #3778

### CODE REVIEW

File: src/content/concepts/module-federation.md [hunk 1]
```diff
@@ -103,13 +104,28 @@ This plugin combines `ContainerPlugin` and `ContainerReferencePlugin`. Overrides
...
 - Control from consumer to container
-    - Overriding modules is a one-directional operation
-    - Sibling containers cannot override each other's modules
-- Concept should be environment-independent
-    - Usable in web, Node.js, etc
+    - Overriding modules is a one-directional operation.
+    - Sibling containers cannot override each other's modules.
+- Concept should be environment-independent.
+    - Usable in web, Node.js, etc.
+- Relative and absolute request in shared.
```
Comment (EugeneHlushko) [inline src/content/concepts/module-federation.md:114]:
```suggestion
- Relative and absolute request in shared:
```

File: src/content/concepts/module-federation.md [hunk 2]
```diff
@@ -103,13 +104,28 @@ This plugin combines `ContainerPlugin` and `ContainerReferencePlugin`. Overrides
...
+    - Will always be provided, even if not used.
+    - Will resolve relative to `config.context`.
+    - Do not use a `requiredVersion` by default.
+- Module requests in shared
+    - are only provided when they are used.
+    - will match all used equal module requests in your build.
+    - will provide all matching modules.
+    - will extract `requiredVersion` from package.json at this position in the graph.
+    - could provide and consume multiple different version when you have nested node_modules.
```
Comment (EugeneHlushko) [inline src/content/concepts/module-federation.md:123]: Why isnt this capitalized? Given the previous list...

File: src/content/concepts/module-federation.md [hunk 3]
```diff
@@ -103,13 +104,28 @@ This plugin combines `ContainerPlugin` and `ContainerReferencePlugin`. Overrides
...
+- Module requests with trailing / in shared
+    - will match all module requests with this prefix.
```
Comment (EugeneHlushko) [inline src/content/concepts/module-federation.md:125]:
```suggestion
- Module requests with trailing `/` in shared will match all module requests with this prefix.
```

File: src/content/concepts/module-federation.md [hunk 4]
```diff
@@ -103,13 +104,28 @@ This plugin combines `ContainerPlugin` and `ContainerReferencePlugin`. Overrides
...
+- New packageName option to choose package name inorder to look for a `requiredVersion`. For module requests it is automatically inferred by default.
```
Comment (EugeneHlushko) [inline src/content/concepts/module-federation.md:126]: I dont see how this belongs to this list, this new option, where does it live? How do you use it? Is it really a part of the concept?
```suggestion
- New `packageName` option allows setting a package name to look for a `requiredVersion`. It is automatically inferred for the module requests by default.
```

File: src/content/concepts/module-federation.md [hunk 5]
```diff
@@ -103,13 +104,28 @@ This plugin combines `ContainerPlugin` and `ContainerReferencePlugin`. Overrides
...
+- `requiredVersion` can now be `false` when automatic infer should be disabled.
```
Comment (EugeneHlushko) [inline src/content/concepts/module-federation.md:127]: This does not belong to concept goals list for sure

File: src/content/concepts/module-federation.md (hunk 4 above)
Comment (jamesgeorge007) [reply src/content/concepts/module-federation.md]: https://github.com/webpack/webpack/blob/c00fec3aa7c79476e5b61651558f5804a48b3bac/schemas/plugins/container/ModuleFederationPlugin.json#L367

File: src/content/concepts/module-federation.md (hunk 5 above)
Comment (jamesgeorge007) [reply src/content/concepts/module-federation.md:126]: Where do you think the respective information would be a best fit?

File: src/content/concepts/module-federation.md [hunk 6]
```diff
@@ -120,3 +134,7 @@ Each page of a Single Page Application is exposed from container build in a sepa
 ### Components library as container
 
 Many applications share a common components library which could be built as a container with each component exposed. Each application consumes components from the components library container. Changes to the components library can be separately deployed without the need to re-deploy all applications. The application automatically uses the up-to-date version of the components library.
+
+### Further reading
```
Comment (EugeneHlushko) [inline src/content/concepts/module-federation.md:138]: Its done via frontmatter at the top of md file

**Comment by EugeneHlushko** [comment]
Thanks!

**Pull request merged**